### PR TITLE
fix(eks): inconsistent fargateProfileName handling causes deletion failure when physicalResourceId exceeds 100 characters

### LINKED ISSUE #32909: [aws-eks]: Inconsistent fargateProfileName Handling Causes Deletion Failure When PhysicalResourceId Exceeds 100 Characters
### Describe the bug

When creating an EKS FargateProfile using the AWS CDK, 
the onCreate event correctly handles long fargateProfileName values by generating a valid name using the generateProfileName() function if none is provided. 

However, in the onDelete event, this logic is missing. 
The deletion process directly uses the physicalResourceId as the fargateProfileName without validating its length. 
If the physicalResourceId exceeds 100 characters, it results in an error during deletion 
because 
1. AWS EKS enforces a maximum length of 100 characters for fargateProfileName.
2. The fargate profile with that name does not exist.

### Regression Issue

- [ ] Select this option if this issue appears to be a regression.

### Last Known Working CDK Version

_No response_

### Expected Behavior

The onDelete event should handle long physicalResourceId values consistently with the creation process. 
and use the same function to generate the fargateProfileName for the delete action

### Current Behavior

Stack deleting failed due to that reason 
 
`Received response status [FAILED] from custom resource. Message returned: The Fargate profile name parameter should not be greater than 100 characters. Logs:`

### Reproduction Steps

1. Deploy a CDK stack that creates a FargateProfile without specifying fargateProfileName, allowing it to auto-generate one.
2. Ensure the generated PhysicalResourceId exceeds 100 characters.
3. Attempt to delete the stack.
4. Observe the deletion failure due to the long fargateProfileName. 

### Possible Solution

Adding the same logic to the onDelete event.

```typescript
protected async onDelete() {
    if (!this.physicalResourceId) {
      throw new Error('Cannot delete a profile without a physical id');
    }

    const fargateProfileName = this.event.ResourceProperties.Config.fargateProfileName ?? this.generateProfileName();
    const deleteFargateProfile: EKS.DeleteFargateProfileCommandInput = {
      clusterName: this.event.ResourceProperties.Config.clusterName,
      fargateProfileName: fargateProfileName,
    };
    this.log({ deleteFargateProfile });
    const deleteFargateProfileResponse = await this.eks.deleteFargateProfile(deleteFargateProfile);
    this.log({ deleteFargateProfileResponse });

    return;
  }
```

### Additional Information/Context

_No response_

### CDK CLI Version

2.175.1

### Framework Version

_No response_

### Node.js Version

22

### OS

MacOS

### Language

TypeScript

### Language Version

_No response_

### Other information

_No response_

### PR DESCRIPTION
**Pull request by newlinedeveloper**
### Issue #

Closes #32909.

### Reason for this change

When deleting an EKS FargateProfile, the `onDelete` event directly uses `physicalResourceId` as the profile name without validating its length. If `physicalResourceId` exceeds 100 characters (AWS EKS's maximum limit), deletion fails with: "The Fargate profile name parameter should not be greater than 100 characters."

The `onCreate` event already handles this by generating a valid name when needed, but this logic was missing in `onDelete`.

### Description of changes

Updated `onDelete()` to use the same name resolution logic as `onCreate()`:
- If `fargateProfileName` is explicitly provided in config: use it
- If `physicalResourceId` is valid (≤ 100 chars): use it directly
- If `physicalResourceId` exceeds 100 chars: generate a name using `generateProfileName()`

**Files changed:**
- `fargate.ts`: Updated `onDelete()` method
- `fargate-resource-provider.test.ts`: Added 5 unit tests covering all scenarios

### Description of how you validated changes

Added comprehensive unit tests:
- Normal deletion with valid `physicalResourceId`
- Explicit `fargateProfileName` handling
- Edge case with `physicalResourceId` > 100 chars
- Explicit name precedence
- Boundary condition (exactly 100 chars)

All tests pass: `yarn test fargate-resource-provider.test.ts`

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)
- [x] Unit tests added/updated
- [x] No breaking changes introduced

---

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by newlinedeveloper** [comment]
**Exemption Request**

Requesting exemption from integration testing requirements for this bug fix:

1. **Bug fix reusing existing logic**: The fix applies the same proven logic from `onCreate()` to `onDelete()`
2. **Comprehensive unit test coverage**: 5 new unit tests cover all scenarios including edge cases
3. **Low risk**: Minimal change that follows existing patterns
4. **Integration test complexity**: Would require creating EKS clusters and reproducing rare edge cases (> 100 char names), which is expensive and difficult to reliably test

The unit tests provide sufficient coverage for this straightforward fix.

**Comment by newlinedeveloper** [comment]
Hi @pahud . Need review for this PR. Thanks

**Comment by pahud** [comment]
> Hi @pahud . Need review for this PR. Thanks

I just clicked CI approval button to trigger the CI run for this PR. After CI passes you should get [pr/pending-community-review](https://github.com/aws/aws-cdk/pulls?q=is%3Apr+is%3Aopen+label%3Apr%2Fneeds-community-review) label and you are all set. Do not click `Update branch` after your PR is ready as this triggers another CI build attempt. Thank you.

**Review by Abogical — DISMISSED** [review]

**Comment by Abogical** [comment]
Sorry, I meant to review your other PR: https://github.com/aws/aws-cdk/pull/35988
Approval retracted.

**Comment by newlinedeveloper** [comment]
Hi @pahud , Need review to this PR. Thanks

File: packages/@aws-cdk/custom-resource-handlers/lib/aws-eks/cluster-resource-handler/fargate.ts [hunk 1]
```diff
@@ -34,9 +34,15 @@ export class FargateProfileResourceHandler extends ResourceHandler {
       throw new Error('Cannot delete a profile without a physical id');
     }
 
+    // Use the same logic as onCreate to determine the profile name
+    // If fargateProfileName was explicitly provided, use it; otherwise use physicalResourceId
+    // If physicalResourceId is too long (> 100 chars), fall back to generating a name
+    const fargateProfileName = this.event.ResourceProperties.Config.fargateProfileName
+      ?? (this.physicalResourceId.length > 100 ? this.generateProfileName() : this.physicalResourceId);
```
Comment (vishaalmehrishi) [inline packages/@aws-cdk/custom-resource-handlers/lib/aws-eks/cluster-resource-handler/fargate.ts:41]: `generateProfileName` uses the current requestId (line 84) to generate a name:

```
const suffix = this.requestId.replace(/-/g, '');
```

If the `requestId` for delete is not the same as the `requestId` for create, the generated profile name will be different and deletion will fail, because it is trying to delete a non-existent profile.

File: packages/@aws-cdk/custom-resource-handlers/lib/aws-eks/cluster-resource-handler/fargate.ts [hunk 2]
```diff
@@ -34,9 +34,15 @@ export class FargateProfileResourceHandler extends ResourceHandler {
       throw new Error('Cannot delete a profile without a physical id');
     }
 
+    // Use the same logic as onCreate to determine the profile name
+    // If fargateProfileName was explicitly provided, use it; otherwise use physicalResourceId
```
Comment (vishaalmehrishi) [inline packages/@aws-cdk/custom-resource-handlers/lib/aws-eks/cluster-resource-handler/fargate.ts]: Based on the logic in `onCreate` (line 25), `physicalResourceId` is the same as the `fargateProfileName`

```
PhysicalResourceId: createFargateProfileResponse.fargateProfile.fargateProfileName,
```

File: packages/@aws-cdk/custom-resource-handlers/lib/aws-eks/cluster-resource-handler/fargate.ts [hunk 3]
```diff
@@ -34,9 +34,15 @@ export class FargateProfileResourceHandler extends ResourceHandler {
...
+    // If physicalResourceId is too long (> 100 chars), fall back to generating a name
+    const fargateProfileName = this.event.ResourceProperties.Config.fargateProfileName
```
Comment (vishaalmehrishi) [inline packages/@aws-cdk/custom-resource-handlers/lib/aws-eks/cluster-resource-handler/fargate.ts:40]: A successful `onCreate` call returns (line 25):

```
PhysicalResourceId: createFargateProfileResponse.fargateProfile.fargateProfileName
```

On a subsequent call (delete), the [constructor sets](https://github.com/newlinedeveloper/aws-cdk/blob/main/packages/%40aws-cdk/custom-resource-handlers/lib/aws-eks/cluster-resource-handler/common.ts#L29):

```
this.physicalResourceId = (event as any).PhysicalResourceId;
```

So when attempting the delete, the `physicalResourceId` is the same as the `fargateProfileName` returned by a successful creation call, implying that the value must be valid.

If the returned `fargateProfileName` name on creation is not valid as a `physicalResourceId` for some reason, the logic should be fixed there instead, since that is the root of the problem.

Updating `onDelete` to try and replicate the ID is not a safe approach and can lead to logic drift - in this specific scenario, the logic here is already out of sync with `queryStatus()`, which checks for `fargateProfileName: this.physicalResourceId` (line 101).

Update/delete operations should always use the `physicalResourceId` from successful creation, unless the intention is to replace - the [CFN documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/crpg-ref.html#crpg-ref-request-fields:~:text=Type%3A%20String-,PhysicalResourceId,-This%20value%20should) explains the behaviour of `PhysicalResourceId ` in detail.

**Review by vishaalmehrishi — CHANGES_REQUESTED** [review]

**Comment by vishaalmehrishi** [comment]
Hi @newlinedeveloper , please take a look at the feedback and let me know if you have any questions. Thank you!
